ttnctl/cmd/devices_register_on_join.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/TheThingsNetwork/go-account-lib/rights"
+	ttnlog "github.com/TheThingsNetwork/go-utils/log"
+	"github.com/TheThingsNetwork/go-utils/random"
+	"github.com/TheThingsNetwork/ttn/api"
+	"github.com/TheThingsNetwork/ttn/api/handler"
+	"github.com/TheThingsNetwork/ttn/api/protocol/lorawan"
+	"github.com/TheThingsNetwork/ttn/core/types"
+	"github.com/TheThingsNetwork/ttn/ttnctl/util"
+	"github.com/spf13/cobra"
+)
+
+const registerOnJoinAccessKeyName = "register-on-join"
+
+var devicesRegisterOnJoinCmd = &cobra.Command{
+	Use:   "on-join [Device ID] [AppKey]",
```

Comment:
Make clear that this is a prefix. Also, be consistent with casing; `DeviceID` or `DevID` (whatever is used elsewhere)